ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -312,6 +312,24 @@ impl Operand {
             _ => todo!(),
         }
     }
+
+    /// Returns the type of the operand.
+    pub(crate) fn type_<'a>(&self, m: &'a Module) -> &'a Type {
+        match self {
+            Self::Local(l) => {
+                match l.instr(m).def_type(m) {
+                    Some(t) => t,
+                    None => {
+                        // When an operand is a local variable, the local can only come from an
+                        // instruction that defines a local variable, and thus has a type. So this
+                        // can't happen if the IR is well-formed.
+                        panic!();
```

Comment:
This looks like an `unreachable` to me?

Reply:
Hah. I almost used `unreachable!()` but talked myself out of it!

I must admit, I get confused as to when to use `todo!()`, `unimplemented!()`, `panic!()` and `unreachable!()`. So I guess `panic!()` suggests an internal consistency, whereas `unreachable!()` suggests you shouldn't be able to get here?

Here it's kind of both. I don't know. We can change it if you want :P

Reply:
They are both very similar: I mildly prefer `unreachable` for the "this case should be impossible" and `panic` for "the caller did something stupid". It's not a big deal though, so use whichever you prefer.

Reply:
https://www.githubstatus.com/

Reply:
Ah, it's appeared now, but the link to it didn't work before.

Let's try again:
Fixed in https://github.com/ykjit/yk/pull/1001/commits/5815c8f03757165a18eb40a0354336a628979b18